### PR TITLE
fix: filter and listMember handler

### LINKED ISSUE #14427: [CAL-3363] User filter on bookings page shows other users in same org even when org has been made private
### Issue Summary

If you create an org and set it to be private, your users in that org should not see other users in the filter

### Steps to Reproduce

1. Create an org
2. Add team members
3. Set the org to private
4. Log in as one of the members and go to filter upcoming/unconfirmed/etc bookings

### Actual Results

- You see all org members

### Expected Results

- You should only see yourself

<sub>[CAL-3363](https://linear.app/calcom/issue/CAL-3363/user-filter-on-bookings-page-shows-other-users-in-same-org-even-when)</sub>

### PR DESCRIPTION
**Pull request by sean-brydon**
fixes: #14427 

Hide people filter 
CleanShot 2024-04-05 at 14 05 14

Also remove members being returned form the org listMember page if its private and youre not an admin. 
This fixes the other place this could leak into the UI in the app OOO page. 
CleanShot 2024-04-05 at 14 06 29

### CODE REVIEW

File: packages/trpc/server/routers/viewer/teams/listMembers.handler.ts [hunk 1]
```diff
@@ -14,6 +14,13 @@ type ListMembersOptions = {
 
 export const listMembersHandler = async ({ ctx, input }: ListMembersOptions) => {
   const { prisma } = ctx;
+  const { isOrgAdmin } = ctx.user.organization;
+  const hasPermsToView = ctx.user.organization.isPrivate ? isOrgAdmin : true;
```
Comment (keithwillcode) [inline packages/trpc/server/routers/viewer/teams/listMembers.handler.ts:18]: Nit: Do we need the ternary?

```suggestion
  const hasPermsToView = !ctx.user.organization.isPrivate || isOrgAdmin;
```

**Review by keithwillcode — DISMISSED** [review]

File: packages/trpc/server/routers/viewer/teams/listMembers.handler.ts (hunk 1 above)
Comment (sean-brydon) [reply packages/trpc/server/routers/viewer/teams/listMembers.handler.ts]: Good spot

File: packages/features/bookings/components/PeopleFilter.tsx [hunk 2]
```diff
@@ -15,6 +15,10 @@ export const PeopleFilter = () => {
   const { t } = useLocale();
   const orgBranding = useOrgBranding();
 
+  const { data: currentOrg } = trpc.viewer.organizations.listCurrent.useQuery();
+  const isAdmin = currentOrg?.user.role === "ADMIN" || currentOrg?.user.role === "OWNER";
+  const hasPermToView = currentOrg?.isPrivate ? isAdmin : true;
```
Comment (keithwillcode) [inline packages/features/bookings/components/PeopleFilter.tsx:20]:
```suggestion
  const hasPermToView = !currentOrg?.isPrivate || isAdmin;
```

**Review by joeauyeung — APPROVED** [review]
Tested PR and people filter no longer renders for members in a private org. The filter still renders for admins or when the privacy setting is turned off.

image

**Pull request merged**